src/components/MarketMySharesForm/index.js

```diff
@@ -27,6 +27,17 @@ class MarketMySharesForm extends Component {
     }
   }
 
+  componentWillMount() {
+    const { gasCosts, gasPrice, requestGasCost, requestGasPrice } = this.props
+
+    if (gasCosts.sellShares === 0) {
+      requestGasCost(GAS_COST.SELL_SHARES)
+    }
+    if (gasPrice.eq(0)) {
```

Comment:
The default should be undefined in the reducer and request the data if it's not available. Inside the components it should fallback to 0 if the gas cost is undefined.

Because if we actually have gas costs that are 0, this would turn into an infinite loop.

Reply:
is set default to 0 to avoid checks in the cost calculations, but yeah, it opens the door to infinite loop in case of gnosis.js returns 0 for any of the costs
This case should never happen, gnosis.js takes costs from a generated json file

Reply:
Maybe we can have a seperate method in the selector that determines if the gas cost was already fetched (e.g. not `undefined`) and a second selector that returns the gas cost and if it does not exist, returns 0 or another default

Reply:
That way you can avoid having to check/cast everywhere before using the gas cost values while still being able to tell with more accuracy, if the values were already fetched.

This could also be solved on a reducer level by introducing new variables that indicate if the gas cost were fetched. What do you think?